client/src/redux/sagas/pullsTestmanSaga.js

```diff
@@ -0,0 +1,27 @@
+import { takeEvery, call, select, put } from 'redux-saga/effects';
+import { PULLS } from '../constants';
+import { fetchTests } from '../api';
+import { setTestman, setTestmanError } from '../actions';
+
+function* handleTestmanLoad() {
+  try {
+    const pulls = yield select(state => state.pulls);
+    const page = 1;
+    const shas = pulls.map(pull => pull.merge_commit_sha);
+    const testResults = yield call(fetchTests, shas[9], shas[0], page);
+    const testByPulls = {};
+    for (let { merge_commit_sha } of pulls) {
```

Comment:
You already have `shas` array here, why iterating over `pulls` and taking the same data?

Reply:
modified